ios/NebulaNetworkExtension/PacketTunnelProvider.swift

```diff
@@ -28,76 +28,96 @@ class PacketTunnelProvider: NEPacketTunnelProvider {
     }
     
     private func start(completionHandler: @escaping (Error?) -> Void) {
-        let proto = self.protocolConfiguration as! NETunnelProviderProtocol
-        var config: Data
-        var key: String
-
-        do {
-            site = try Site(proto: proto)
-            config = try site!.getConfig()
-        } catch {
-            //TODO: need a way to notify the app
-            log.error("Failed to render config from vpn object")
-            return completionHandler(error)
-        }
+        // Load vpn configs from system, and find the manager matching the one being started
+        NETunnelProviderManager.loadAllFromPreferences() { managers, error in
+            let targetProtoConfig = self.protocolConfiguration as? NETunnelProviderProtocol;
+            let targetID = targetProtoConfig?.providerConfiguration!["id"] as? String;
+            
+            guard let managers = managers else {
+                self.log.error("No managers were loaded")
+                return
+            }
+            
+            for manager in managers {
+                var config: Data
+                var key: String
+                
+                let mgr = manager.protocolConfiguration as? NETunnelProviderProtocol;
+                let id = mgr?.providerConfiguration!["id"] as? String;
+                
+                if (id == targetID) {
```

Comment:
If we find our `id` then we can stop looping and finish starting, we can close this loop a lot sooner and avoid the `break` at the bottom